doc/design/doi-generation-2.md

```diff
@@ -0,0 +1,229 @@
+      - publish dandiset
+
+### Issues with the Existing Approach
+
+- [Stop injecting "fake" DOIs into draft dandisets](https://github.com/dandi/dandi-archive/issues/1709)
+- [Unpublished Dandisets display a DOI under `Cite As`](https://github.com/dandi/dandi-archive/issues/1932)
+
+## Background
+
+Initially proposed/discussed in
+
+- [Create and maintain a "Findable" DOI for the Dandiset as a whole](https://github.com/dandi/dandi-archive/issues/1319)
+
+and boils down to the adoption of approach of Zenodo of having a DOI which always points to the latest version of the record.
+[Zenodo uses the language](https://support.zenodo.org/help/en-gb/1-upload-deposit/97-what-is-doi-versioning) `Concept DOI` to mean a top-level DOI that references all versions, which we will refer to as `Dandiset DOI`.
+
+DataCite allows for three types of DOIs ([DataCite](https://support.datacite.org/docs/what-does-the-state-of-the-doi-mean)):
+
+- `Draft`. We do not use those.
+  *Can be deleted, and they require only the identifier itself in order to be created or saved. They can be updated to either Registered or Findable DOIs. Registered and Findable DOIs may not be returned to the Draft state, which means that changing the state of a Draft is final.*
+- `Registered`. Like `Findable` but not indexed for search, so we do not use them.
+- `Findable`. Is the type we use for published dandisets.
+  Requires to be valid (pass validation to fit the datacite schema) to be created.
+
+## Proposed Solution
+
+- For **Public dandisets**:
+    - Upon creation:
+        - mint a `Dandiset DOI` (a DataCite `Draft DOI`) `10.48324/dandi.{dandiset.id}` with *minimal metadata* entered during creation request (title, description, license)
+        - URL should point be DLP `https://dandiarchive.org/dandiset/{dandiset.id}`
+        - If minting a DOI fails, we need to raise exception to inform developers about the issue but proceed with the creation of the dandiset.
+    - Upon updates to a draft dandiset metadata **prior to first publication**:
+        - Update the datacite metadata of the `Draft DOI`, (leave as draft)
+        - If validation fails, log error and continue
+    - Upon deletion of a draft dandiset metadata **prior to first publication**:
+        - Delete the `Dandiset DOI` (Draft) from Datacite
+    - Upon **first publication** of a dandiset:
+        - Mint a new `Version DOI` (Findable) (already done currently), ie `10.48324/dandi.{dandiset.id}/{version}`
+        - Update `Dandiset DOI` metadata to match published version
+        - promote `Dandiset DOI` (Draft) to `Findable DOI`
+    - Upon updates to draft dandiset metadata **after the first publication**"
+        - no-op. The `Dandiset DOI` metadata will match the most recent publication.
+    - Upon deletion of a draft dandiset metadata **after the first publication**:
+        - "hide" the `Dandiset DOI` (Findable) to `Registered DOI`
```

Comment:
```suggestion
    - Upon deletion of a published dandiset version (`VersionViewSet.destroy`) :
        - "hide" the `Version DOI` (Findable) to `Registered DOI`
    - Upon deletion of a dandiset (`DandisetViewSet.destroy`):
        - "hide"  the `Dandiset DOI`  if `Findable` and delete if `Draft`
```

note also that currently `delete_doi` which is invoked from within `VersionViewSet.destroy` seems to be useless since can delete only Draft ones, but this is for published version, so cannot do that